net-utils/src/lib.rs

```diff
@@ -519,31 +519,44 @@ pub fn bind_common(
         .and_then(|_| TcpListener::bind(&addr).map(|listener| (sock.into(), listener)))
 }
 
-pub fn bind_two_consecutive_in_range(
+pub fn bind_two_in_range_with_offset(
     ip_addr: IpAddr,
     range: PortRange,
+    offset: u16,
 ) -> io::Result<((u16, UdpSocket), (u16, UdpSocket))> {
+    if range.1.saturating_sub(range.0) < offset {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            "range too small to find two ports with the correct offset".to_string(),
+        ));
+    }
     let mut first: Option<UdpSocket> = None;
     for port in range.0..range.1 {
+        if let Some(first_bind) = &first {
+            if port.saturating_sub(first_bind.local_addr().unwrap().port()) < offset {
+                continue;
+            }
+        }
         if let Ok(bind) = bind_to(ip_addr, port, false) {
-            match first {
-                Some(first_bind) => {
+            if let Some(first_bind) = first {
```

Comment:
How about this?
For each port in the range, try binding port as the first socket. If successful try binding port + offset as the second socket. If successful return. Else continue..

Reply:
Like it. Pushed those changes.